### PR TITLE
Added startDateRequired and endDateRequired inputs to date range picker

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**

### CODE REVIEW

File: src/app/public/modules/datepicker/datepicker.component.ts [hunk 1]
```diff
@@ -202,11 +202,18 @@ export class SkyDatepickerComponent implements OnDestroy, OnInit {
...
     const uniqueId = nextId++;
     this.calendarId = `sky-datepicker-calendar-${uniqueId}`;
     this.triggerButtonId = `sky-datepicker-button-${uniqueId}`;
+
+    // Update icons when theme changes.
+    themeSvc?.settingsChange
+    .pipe(takeUntil(this.ngUnsubscribe))
+    .subscribe(() => {
+      this.changeDetector.markForCheck();
+    });
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/datepicker/datepicker.component.ts:216]: ☝️ Not in the scope of this PR, but I upgrading theme revealed a bug. Icons weren't updating when the user switches themes back and forth. It was caught in our unit tests, so it needed fixed for this work to continue. This fix ensures we force change detection to fire when the theme switches.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-date-range-required](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-date-range-required)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-date-range-required](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-date-range-required)

File: src/app/public/modules/date-range-picker/date-range-picker.component.ts [hunk 2]
```diff
@@ -251,14 +267,22 @@ export class SkyDateRangePickerComponent
...
     this.localeProvider.getLocaleInfo()
       .pipe(takeUntil(this.ngUnsubscribe))
       .subscribe((localeInfo) => {
         SkyDateFormatter.setLocale(localeInfo.locale);
         this.preferredShortDateFormat = SkyDateFormatter.getPreferredShortDateFormat();
       });
+
+    // Update icons when theme changes.
+    themeSvc?.settingsChange
+    .pipe(takeUntil(this.ngUnsubscribe))
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/date-range-picker/date-range-picker.component.ts:282]: Tabs.

File: src/app/public/modules/date-range-picker/date-range-picker.component.ts [hunk 3]
```diff
@@ -64,6 +65,7 @@ import {
 import {
   SkyDateRangeService
 } from './date-range.service';
+import { SkyThemeService } from '@skyux/theme';
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/date-range-picker/date-range-picker.component.ts:68]: Import style.

File: src/app/public/modules/date-range-picker/date-range-picker.component.spec.ts [hunk 4]
```diff
@@ -32,6 +32,9 @@ import {
 } from './types/date-range-calculator-type';
 
 import * as moment_ from 'moment';
+import { SkyTheme, SkyThemeMode, SkyThemeService, SkyThemeSettings, SkyThemeSettingsChange } from '@skyux/theme';
+import { BehaviorSubject } from 'rxjs';
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/date-range-picker/date-range-picker.component.spec.ts:36]: Import style.

File: src/app/public/modules/datepicker/datepicker.component.ts [hunk 5]
```diff
@@ -202,11 +202,18 @@ export class SkyDatepickerComponent implements OnDestroy, OnInit {
...
-    @Optional() public themeSvc?: SkyThemeService
+    @Optional() themeSvc?: SkyThemeService
   ) {
     const uniqueId = nextId++;
     this.calendarId = `sky-datepicker-calendar-${uniqueId}`;
     this.triggerButtonId = `sky-datepicker-button-${uniqueId}`;
+
+    // Update icons when theme changes.
+    themeSvc?.settingsChange
+    .pipe(takeUntil(this.ngUnsubscribe))
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/datepicker/datepicker.component.ts:213]: Tabs.

File: src/app/public/modules/timepicker/timepicker.component.ts [hunk 6]
```diff
@@ -254,11 +254,18 @@ export class SkyTimepickerComponent implements OnInit, OnDestroy {
...
-    @Optional() public themeSvc?: SkyThemeService
+    @Optional() themeSvc?: SkyThemeService
   ) {
     const uniqueId = nextId++;
     this.timepickerId = `sky-timepicker-${uniqueId}`;
     this.triggerButtonId = `sky-timepicker-button-${uniqueId}`;
+
+    // Update icons when theme changes.
+    themeSvc?.settingsChange
+    .pipe(takeUntil(this.ngUnsubscribe))
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/timepicker/timepicker.component.ts:265]: Tabs.

File: src/app/public/modules/date-range-picker/date-range-picker.component.ts [hunk 7]
```diff
@@ -178,6 +180,20 @@ export class SkyDateRangePickerComponent
   @Input()
   public label: string;
 
+  /**
+   * Indicates whether to mark the start date input as required.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/date-range-picker/date-range-picker.component.ts:184]:
```suggestion
   * Indicates whether to require users to specify a start date.
```
I assume this property actually makes the property required and doesn't just visually mark it as required.

File: src/app/public/modules/date-range-picker/date-range-picker.component.ts [hunk 8]
```diff
@@ -178,6 +180,20 @@ export class SkyDateRangePickerComponent
...
+   * @default false
+   */
+  @Input()
+  public startDateRequired: boolean = false;
+
+  /**
+   * Indicates whether to mark the end date input as required.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/date-range-picker/date-range-picker.component.ts:191]:
```suggestion
   * Indicates whether to require users to specify an end date.
```
Same assumption as for start date.

File: src/app/public/modules/date-range-picker/date-range-picker.component.ts (hunk 7 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/date-range-picker/date-range-picker.component.ts]: Done.

File: src/app/public/modules/date-range-picker/date-range-picker.component.ts (hunk 8 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/date-range-picker/date-range-picker.component.ts]: Done.

File: src/app/public/modules/datepicker/datepicker.component.ts (hunk 5 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/datepicker/datepicker.component.ts]: Fixed.

File: src/app/public/modules/timepicker/timepicker.component.ts (hunk 6 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/timepicker/timepicker.component.ts]: Fixed.

File: src/app/public/modules/date-range-picker/date-range-picker.component.ts (hunk 2 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/date-range-picker/date-range-picker.component.ts]: Fixed.

File: src/app/public/modules/date-range-picker/date-range-picker.component.spec.ts (hunk 4 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/date-range-picker/date-range-picker.component.spec.ts]: Fixed.

File: src/app/public/modules/date-range-picker/date-range-picker.component.ts (hunk 3 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/date-range-picker/date-range-picker.component.ts]: Fixed.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-date-range-required](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-date-range-required)

**Pull request merged**